scripts/build-macos.sh

```diff
@@ -1,41 +1,43 @@
 #!/bin/bash
 
-export GOOS=darwin 
+export GOOS=darwin
 export GOARCH=amd64
 
+PROJECT_ROOT="$(dirname "$(dirname "$(readlink -f "$0")")")"
+cd "${PROJECT_ROOT}" || exit 1
+
 echo "building swapd..."
-cd cmd/daemon 
-if ! go build -o swapd-amd64-darwin ; then
+cd cmd/daemon || exit 1
+if ! go build -o swapd-amd64-darwin; then
 	exit 1
 fi
 mv swapd ../..
 echo "done building swapd."
 
 echo "building swapcli..."
-cd ../client 
-if ! go build -o swapcli-amd64-darwin ; then 
+cd ../client || exit 1
+if ! go build -o swapcli-amd64-darwin; then
 	exit 1
 fi
 mv swapcli ../..
 echo "done building swapcli."
 
 if [[ -z "${ALL}" ]]; then
-  exit 0
+	exit 0
 fi
 
 echo "build swaprecover..."
-cd ../recover
-if ! go build -o swaprecover-amd64-darwin ; then 
+cd ../recover || exit 1
```

Comment:
It was either add the `exit 1`, or add a comment to tell the linter to ignore it.  None of these build scripts are sourced (verses executed), so the exit should be safe.  Most of these lines will go away if we do the directory renames that I mentioned above.